flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointOptions.java

```diff
@@ -179,6 +254,16 @@ public CheckpointStorageLocationReference getTargetLocation() {
         return targetLocation;
     }
 
+    /**
+     * Returns whether the snapshot needs to be a full one in the sense if it can share artefacts
+     * with prior snapshots.
+     *
+     * @return If it is forced to be a full snapshot it should not share any artefacts.
+     */
+    public boolean isForcedFullSnapshot() {
+        return forcedFullSnapshot;
+    }
```

Comment:
In the context of https://issues.apache.org/jira/browse/FLINK-25276 I've realised that "force full snapshot" will not be enough. For incremental savepoints we will actually need a yet another different operating mode. In total we would have:

1. (current master) you can re-use the artefacts as you wish
2. "forceFullSnapshot"/"no-claim" mode, you can not re-use old artefacts, but the newly uploaded artefacts in this snapshot can be re-used in the future
3. "incremental savepoint" mode, you can not re-use old artefacts and newly uploaded artefacts can not be re-used in the future

So I would propose to change this `boolean` flag to `enum`. In this PR this enum would have two values (1. and 2.). For FLINK-25192, we would add the 3rd mode. I would consider putting this enum inside `CheckpointType`?

Reply:
Makes sense. I updated it as you described it, ptal.